ArmPkg/Library/ArmFfaLib/ArmFfaStandaloneMmCoreLib.inf

```diff
@@ -0,0 +1,36 @@
+## @file
+#  Provides FF-A ABI Library used in StandaloneMmCore.
+#
+#  Copyright (c) 2024, Arm Limited. All rights reserved.<BR>
+#
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+[Defines]
+  INF_VERSION                    = 0x0001001B
+  BASE_NAME                      = ArmFfaStandaloneMmCoreLib
+  FILE_GUID                      = 80d2c4dc-5f0b-11ef-bc86-43b3fb486d6d
+  MODULE_TYPE                    = MM_CORE_STANDALONE
+  VERSION_STRING                 = 1.0
+  PI_SPECIFICATION_VERSION       = 0x00010032
+  LIBRARY_CLASS                  = ArmFfaLib
+  CONSTRUCTOR                    = ArmFfaStandaloneMmLibConstructor
+
+[Sources]
+  ArmFfaCommon.h
+  ArmFfaCommon.c
+  ArmFfaStandaloneMmLib.c
+
+[Packages]
+  MdePkg/MdePkg.dec
+  ArmPkg/ArmPkg.dec
+
+[LibraryClasses]
+  ArmSmcLib
+  ArmSvcLib
+  BaseLib
+  BaseMemoryLib
+  DebugLib
+
+[FixedPcd]

```

Comment:
While I was testing this, this PCD seems tricky. The PCD per module override will not take effect, which will be needed for DXE vs. Standalone MM. This will cause some silent PCD override issues that even build report will not tell us.

Can we change it in the inf files to be not fixed pcd, and in code we do not take fixed PCD either? In dec, we can move it to the fixed/dyn/patch section?

Reply:
> While I was testing this, this PCD seems tricky. The PCD per module override will not take effect, which will be needed for DXE vs. Standalone MM. This will cause some silent PCD override issues that even build report will not tell us.
> 
> Can we change it in the inf files to be not fixed pcd, and in code we do not take fixed PCD either? In dec, we can move it to the fixed/dyn/patch section?

> While I was testing this, this PCD seems tricky. The PCD per module override will not take effect, which will be needed for DXE vs. Standalone MM. This will cause some silent PCD override issues that even build report will not tell us. 

Why is this PCD per module override is required? Actually, I want to determined the value of PCD only in dsc file
So that, when StandaloneMm build. All moulde should be used Svc conduit by setting it false.
otherwise DXE driver (in UEFI) using only for Smc conduit.

Reply:
That was because not necessarily all mm modules have to be in a separate DSC file :) It is unfortunate that UEFI does not support per module type PCD assignments, which will lead to per module override if stmm modules sit together with other module types.

I agree that your original use case works. But I think the fact that per module override does not work makes it very hard to debug.